src/main/java/com/avioconsulting/mule/opentelemetry/internal/store/InMemoryTransactionStore.java

```diff
@@ -142,6 +142,8 @@ public TransactionMeta endTransaction(
     }
     if (transactionMeta == null) {
       LOGGER.trace("No transaction meta found for {} ", traceComponent);
+    } else if (transactionMeta.getTags() != null && transactionMeta.getTags() != null) {
```

Comment:
Is there a reason we are checking the same values twice for null?

Reply:
One was supposed to be tracecomponent :). I don't think both will ever be null but for safety check. Fixed in latest commit.